editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -4384,11 +4387,35 @@ void CanvasItemEditor::_popup_callback(int p_op) {
 
 	last_option = MenuOption(p_op);
 	switch (p_op) {
+		case TOGGLE_GRID: {
+			int prev_grid_visibility_op = grid_visibility_op;
+
+			switch (prev_grid_visibility_op) {
+				case ALWAYS_SHOW_GRID: {
+					grid_visibility_op = ALWAYS_HIDE_GRID;
```

Comment:
We should probably go to the least impactful option here. So basically:
```
grid_visibility_op = grid_snap_active ? ALWAYS_HIDE_GRID : SHOW_GRID_WHEN_SNAPPING_TO_GRID;
```

Reply:
Are you suggesting that I replace the entire switch block with that line?, I don't think that would work, toggle grid is supposed to always toggle the grid, if I replace the switch with that line, if the user attempts to show the grid using the shortcut key when grid snapping is disabled, the grid will not be shown.

Reply:
No, only the case for ALWAYS_SHOW_GRID. This aims to always go to the least restrictive option possible that toggles the grid.